File: src/ControleBO.Domain/Models/ObjetoApreendido.cs
using ControleBO.Domain.Core.Models;
using System;

namespace ControleBO.Domain.Models
{
    public class ObjetoApreendido : Entity
    {
        public ObjetoApreendido(string descricao, string local, Procedimento procedimento, DateTime? dataApreensao)
        {
            Descricao = descricao;
            Local = local;
            Procedimento = procedimento;
            DataApreensao = dataApreensao;
        }

        public ObjetoApreendido(int id, string descricao, string local, Procedimento procedimento, DateTime? dataApreensao)
            : this(descricao, local, procedimento, dataApreensao)
        {
            Id = id;
        }

        protected ObjetoApreendido() { }

        public string Descricao { get; set; }

        public string Local { get; set; }

        public DateTime? DataApreensao { get; set; }

        public int ProcedimentoId { get; set; }

        public virtual Procedimento Procedimento { get; set; }
    }
}

File: src/ControleBO.AngularDesktop/src/app/models/objeto-apreendido.ts
export class ObjetoApreendido {
  id: number;
  descricao: string;
  local?: string;
  dataInsercao: Date = new Date();
  dataApreensao: Date;
  procedimentoId: number;

  get DataInsercao() {
    return this.dataInsercao;
  }

  set DataInsercao(value: any) {
    if (value) {
      this.dataInsercao = new Date(value);
    }
  }

  get DataApreensao() {
    return this.dataApreensao;
  }

  set DataApreensao(value: any) {
    if (value) {
      this.dataApreensao = new Date(value);
    }
  }

  constructor(procedimentoId: number, obj?: any) {
    if (obj) {
      Object.assign(this, obj);
      this.DataApreensao = obj.dataApreensao;
    }
    this.procedimentoId = procedimentoId;
  }
}
